### PR TITLE
Add methods to create a sub-OSMGraph from an OSMGraph

### PR DESCRIPTION
**Pull request by bjmommers**
Create an OSMGraph that is a subgraph of the original with the relevant metadata. Requires an OSMGraph and a list of vertices or node IDs.

### CODE REVIEW

File: src/graph.jl [hunk 1]
```diff
@@ -452,3 +452,81 @@ function add_kdtree!(g::OSMGraph)
...
+                 )::OSMGraph where {U <: Integer, T <: Integer, W <: Real}
+
+Create an OSMGraph representing a subgraph of another OSMGraph containing 
+specified vertices.
+The resulting OSMGraph object will also contain vertices from all ways that the
+specified vertices (nodes) are members of.
+"""
+function osm_subgraph(g::OSMGraph{U, T, W},
+                      vertex_list::Vector{U}
+                     ) where {U, T, W}
```
Comment (captchanjack) [inline src/graph.jl:468]:
```suggestion
                      ) where {U, T, W}
```

File: src/graph.jl [hunk 2]
```diff
@@ -452,3 +452,81 @@ function add_kdtree!(g::OSMGraph)
     cartesian_locations = to_cartesian(node_locations)
     g.kdtree = KDTree(cartesian_locations)
 end
+
```
Comment (captchanjack) [inline src/graph.jl:455]: thinking maybe we move this to its own file, "subgraph.jl"

File: src/LightOSM.jl [hunk 3]
```diff
@@ -31,6 +31,7 @@ export GeoLocation,
        graph_from_object,
        graph_from_download,
        graph_from_file,
+       osm_subgraph,
```
Comment (captchanjack) [inline src/LightOSM.jl:34]: unit tests!

Comment (captchanjack) [inline src/LightOSM.jl:34]: Also can we add these to the /docs dir as well?

File: src/graph.jl (hunk 1 above)
Comment (bjmommers) [reply src/graph.jl]: Fixed, included with other changes.

File: src/LightOSM.jl (hunk 3 above)
Comment (bjmommers) [reply src/LightOSM.jl]: Tests added, thanks for the reminder :)

File: src/graph.jl (hunk 2 above)
Comment (bjmommers) [reply src/graph.jl:455]: Created "subgraph.jl" and moved `osm_subgraph` into it. Module and unit tests updated to reflect the new file.

File: src/LightOSM.jl (hunk 3 above)
Comment (bjmommers) [reply src/LightOSM.jl]: Added `osm_subgraph` under "create_graph.md" in the docs.

**Pull request merged**